galflow/python/shear.py

```diff
@@ -0,0 +1,69 @@
+# Functions computing shear related affine transformations
+# based on https://github.com/GalSim-developers/GalSim/blob/main/galsim/shear.py
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+from galflow.python.transform import transform
+
+__all__ = ["shear", "shear_transformation"]
+
+def shear_transformation(g1, g2, Fourier=False):
+  """
+  Function to compute the affine transformation corresponding to a given shear.
+  This function uses the reduced shear definition:
+
+    :math:`|g| = \frac{a - b}{a + b}`
+
+  TODO: better documentation (the following is stolen from galsim)
+
+  If a field is sheared by some shear, s, then the position (x,y) -> (x',y')
+  according to:
+  .. math::
+      \left( \begin{array}{c} x^\prime \\ y^\prime \end{array} \right)
+      = S \left( \begin{array}{c} x \\ y \end{array} \right)
+  and :math:`S` is the return value of this function ``S = shear.getMatrix()``.
+  Specifically, the matrix is
+  .. math::
+      S = \frac{1}{\sqrt{1-g^2}}
+              \left( \begin{array}{cc} 1+g_1 & g_2 \\
+                                       g_2 & 1-g_1 \end{array} \right)
+  Parameters
+  ----------
+    g1: Tensor
+      The first component of the shear in the "reduced shear" definition.
+    g2: Tensor
+     The second component of the shear in the "reduced shear" definition.
+
+  Returns
+  -------
+    Transformation matrix format, i.e [(batch), 3, 3]
```

Comment:
use proper docstring formatting

Reply:
I'm going to let it fly for now because we are going to change the documentation style anyway